### DISCUSSION TITLE
Authentication DICOM Web server - OAuth 2.0

### DISCUSSION THREAD

**Discussion by nicholasbuenger** [Project-MONAI/MONAILabel#1539] [Q&A] [answered]
Hello,

I would like to authenticate to a DICOM Web server with OAuth 2.0 standard.
I am aware of the authentication with `MONAI_LABEL_DICOMWEB_USERNAME `and `MONAI_LABEL_DICOMWEB_PASSWORD`, but OAuth 2.0 would better. Is it possible do use it? 

Cheers,
Nicholas

**Comment by SachidanandAlle** [comment, marked as answer]
When we start MONAI Label server, we don't receive any remote request.  
We need to connect to DICOM Web even before receiving any user requests.

However, many DicomWeb servers allow host whitelist.  In that case you can add MONAILabel server to whitelist in DICOMWeb server (example in Orthanc config) and that way you don't need to share the username/password to MONAILabel server.

Or may be we can accept some Token and pass `Authorization` header (from env variable) in DICOMWeb requests instead of Username and Password.  But such tokens are normally short-lived and MONAI Server might be up and running for days/weeks.  It will become a hassle to refresh the token everytime.

**Reply by nicholasbuenger** [reply]
Thank you
